### PR TITLE
Correctly respond to fontSettings and allFontSettings

### PR DESCRIPTION
**Pull request by timalacey**
spelling

### CODE REVIEW

**Comment by timalacey** [comment]
@TwitchBronBron can you review?
Thanks

File: test-project/components/MainScene.brs [hunk 1]
```diff
@@ -57,31 +57,45 @@ end function
...
-    }
-    italicFont: {
-      name: "font:MediumBoldSystemFont"
-      size: 32
-    }
-    boldFont: {
-      name: "font:MediumSystemFont"
-      size: 34
-  } }
+  ' m.label.allFontSettings = {
```
Comment (TwitchBronBron) [inline test-project/components/MainScene.brs]: Why are these commented out? If they aren't needed anymore, then we should delete the code.

**Comment by TwitchBronBron** [comment]
@timalacey can you add a little more explanation as to what the purpose of some of the other changes were for? It looks to be a lot more than spelling changes at this point. ;)

File: test-project/components/MainScene.brs (hunk 1 above)
Comment (timalacey) [reply test-project/components/MainScene.brs]: Cleaned up.

**Comment by timalacey** [comment]
> @timalacey can you add a little more explanation as to what the purpose of some of the other changes were for? It looks to be a lot more than spelling changes at this point. ;)

Sorry that was a result of PR-creep. I spotted the spelling error, then discovered fontSettings and allFontSettings weren't responding per the README.
I've updated the PR name.

File: src/components/SmartLabel.brs [hunk 2]
```diff
@@ -147,13 +149,14 @@ function updateFonts(event = invalid)
   if event = invalid
     fontNames = m.fontFields
   else
-    fontNames = [event.getField()]
+    field = event.getField()
+    fontNames = [field.replace("Settings", "")]
   end if
```
Comment (georgejecook) [inline src/components/SmartLabel.brs:155]: I'm not a fan of this change. We're iterating over **names** in the `fontNames` collection. Even if these _were_ styles, then the name would be `styleName`; not `style`.

File: src/components/SmartLabel.brs [hunk 3]
```diff
@@ -147,13 +149,14 @@ function updateFonts(event = invalid)
...
 
-  for each name in fontNames
-    fontSettings = m.top[name + "Settings"]
```
Comment (georgejecook) [inline src/components/SmartLabel.brs:154]: why change this to something less specific? Don't get this change. `fontSettings` is clearer than `settings` here.

File: src/components/SmartLabel.brs [hunk 4]
```diff
@@ -147,13 +149,14 @@ function updateFonts(event = invalid)
   if event = invalid
     fontNames = m.fontFields
   else
-    fontNames = [event.getField()]
+    field = event.getField()
+    fontNames = [field.replace("Settings", "")]
```
Comment (georgejecook) [inline src/components/SmartLabel.brs:153]: I'm probably naive. What's the point of this change?

File: src/components/SmartLabel.brs [hunk 5]
```diff
@@ -185,28 +188,31 @@ function makeFontSetting(name, size, color = "#000000")
...
-    if (type(setting) = "roAssociativeArray")
-      if m.settingsKeys[key] <> invalid
-        m[style] = setting
+function onAllFontSettingsChange(event as Object)
+  settings = event.getData()
+  for each style in settings
+    setting = settings[style]
+    if type(setting) = "roAssociativeArray"
+      if m.settingsKeys[style + "Settings"] <> invalid
+        m[style] = getFont(setting.name, setting.size)
```
Comment (georgejecook) [inline src/components/SmartLabel.brs]: > this is wrong. m[style] contains an aa style - here we've changed it to contain the font. Please read the xml comment that goes with the `allFontSettings` field. i.e

Okay - ignore that!! I totally forgot that I'd changed how this works, and badly (hence your bugs). sorry.. this change is correct :)

File: test-project/components/MainScene.brs [hunk 6]
```diff
@@ -57,31 +57,16 @@ end function
...
-    }
-    italicFont: {
-      name: "font:MediumBoldSystemFont"
-      size: 32
-    }
-    boldFont: {
-      name: "font:MediumSystemFont"
-      size: 34
-  } }
+  m.label.fontSettings = {
```
Comment (georgejecook) [inline test-project/components/MainScene.brs]: I concur with showing the color setting usage here; but I'd like to not remove the other settings, as the point was to show how one configures all font settings in one place.

**Review by georgejecook — CHANGES_REQUESTED** [review]
Thanks for taking the time to contribute!! :)

There's a couple of notes of things that I don't understand. I'm hoping you will be interested in further contributing, as I chucked this together in a couple of hours as a demo for ropm; but I think it's got much wider use, with a bit of love.. you may have noticed that there are issues with the heading settings and wrapping is not supported. I'd very much welcome contributions addressing those issues, if you are interested in bringing this forward.
Thanks again!!

File: src/components/SmartLabel.brs (hunk 4 above)
Comment (timalacey) [reply src/components/SmartLabel.brs:153]: `m.fontFields = ["font", "header3Font", "header2Font", "header1Font", "italicFont", "boldFont"]`

`event.getField()` can return "fontSettings", "header1FontSettings", "header2FontSettings", "header3FontSettings", "italicFontSettings", "boldFontSettings"

File: src/components/SmartLabel.brs (hunk 5 above)
Comment (timalacey) [reply src/components/SmartLabel.brs]: This change is to align with `updateFonts()`

File: src/components/SmartLabel.brs (hunk 2 above)
Comment (timalacey) [reply src/components/SmartLabel.brs:155]: Updated

File: src/components/SmartLabel.brs (hunk 3 above)
Comment (timalacey) [reply src/components/SmartLabel.brs:154]: Updated

**Review by timalacey — COMMENTED** [review]
Updated, ready for review.

**Comment by georgejecook** [comment]
@TwitchBronBron looks good to me mate. lgtm.

Thanks @timalacey !!! :star:

**Review by georgejecook — APPROVED** [review]
thanks for the changes, and clarifications.

**Pull request merged**